src/Dapr.Actors/Runtime/Actor.cs

```diff
@@ -20,41 +20,25 @@ namespace Dapr.Actors.Runtime
     public abstract class Actor
     {
         private const string TraceType = "Actor";
-        private readonly string traceId;
-        private readonly string actorImplementaionTypeName;
+        private string traceId;
+        private string actorImplementationTypeName;
 
         /// <summary>
         /// Contains timers to be invoked.
         /// </summary>
         private readonly Dictionary<string, IActorTimer> timers = new Dictionary<string, IActorTimer>();
 
-        /// <summary>
-        /// Initializes a new instance of the <see cref="Actor"/> class.
-        /// </summary>
-        /// <param name="actorService">The <see cref="ActorService"/> that will host this actor instance.</param>
-        /// <param name="actorId">Id for the actor.</param>
-        /// <param name="actorStateManager">The custom implementation of the StateManager.</param>
-        protected Actor(ActorService actorService, ActorId actorId, IActorStateManager actorStateManager = default)
-        {
-            this.Id = actorId;
-            this.traceId = this.Id.ToString();
-            this.IsDirty = false;
-            this.ActorService = actorService;
-            this.StateManager = actorStateManager ?? new ActorStateManager(this);
-            this.actorImplementaionTypeName = this.ActorService.ActorTypeInfo.ImplementationType.Name;
-        }
-
         /// <summary>
         /// Gets the identity of this actor.
         /// </summary>
         /// <value>The <see cref="ActorId"/> for the actor.</value>
-        public ActorId Id { get; }
+        public ActorId Id { get; private set; }
```

Comment:
How would user provide its own constructor for Actor? In its constructor, ActorId will not be available>

Reply:
I think there are several ways for that:

- Make the setter for ActorId public - I wouldn't do that
- ActorId is currently set in OnActivateInternalAsync() - make that method public
- Extract the code that was originally in the Actor() constructor from OnActivateInternalAsync() to a public method ConfigureActor(actorId, actorService, ...)

I personally would go for option 3.

Reply:
I know this is kind of "not object oriented programming" because the Actor would not be fully instantiated by the constructor but would need an extra ConfigureActor() call.
But the problem with this basic kind of IoC container IServiceProvider implements, that AFAIK you can't easily inject instance runtime information (like an ActorId).

Reply:
User does not know the actor id, it needs to be provided in the constructor when a call comes for it, OnActivateAsync seems too late in the actor lifecyle.

Reply:
OK, I have to differentiate:

- In normal (DAPR) operation, the Actor is instantiated and activated in the ActorManager:
```
        internal async Task ActivateActor(ActorId actorId)
        {
            // An actor is activated by "Dapr" runtime when a call is to be made for an actor.
            var actor = this.actorService.CreateActor();
            await actor.OnActivateInternalAsync(this.actorService, actorId);
        ...
        }
```
So there's practically no difference between providing the ActorService and ActorId in the constructor or the call to OnActivateInternalAsync(). So we also could easily have the instantiation and activation like this:
```
        internal async Task ActivateActor(ActorId actorId)
        {
            // An actor is activated by "Dapr" runtime when a call is to be made for an actor.
            var actor = this.actorService.CreateActor();
            actor.Configure(this.actorService, actorId);
            await actor.OnActivateInternalAsync();
        ...
        }
```

- For unit-testing the user has to provide the ActorService (or a mock thereof) and an ActorId. So for that we need same public methode like actor.Configure(...)

Reply:
Let me think more about it, when the user actor constructor is called, they can choose to load  things based on the actor id.
IMO, public Configure method for mocking doesnt seem like the right thing to do.